packages/payment/src/services/payment-provider.ts

```diff
@@ -0,0 +1,238 @@
+
+    const error = await provider.deletePayment(paymentSession.data)
+    if (isPaymentProcessorError(error)) {
+      this.throwFromPaymentProcessorError(error)
+    }
+  }
+
+  async createPayment(
+    data: CreatePaymentInput
+  ): Promise<Record<string, unknown>> {
+    const { payment_session, provider_id } = data
+    const providerId = provider_id ?? payment_session.provider_id
+
+    const provider = this.retrieveProvider(providerId)
+
+    /**
+     * NOTE: JUST RETRIEVE
+     */
+    const paymentData = await provider.retrievePayment(payment_session.data)
+    if (isPaymentProcessorError(paymentData)) {
+      this.throwFromPaymentProcessorError(paymentData)
+    }
+
+    return paymentData as Record<string, unknown>
+  }
+
+  async authorizePayment(
+    paymentSession: {
+      provider_id: string
+      data: Record<string, unknown>
+    },
+    context: Record<string, unknown>
+  ): Promise<{ data: Record<string, unknown>; status: PaymentSessionStatus }> {
+    const provider = this.retrieveProvider(paymentSession.provider_id)
+
+    const res = await provider.authorizePayment(paymentSession.data, context)
+    if (isPaymentProcessorError(res)) {
+      this.throwFromPaymentProcessorError(res)
+    }
+
+    const { data, status } = res as PaymentProcessorAuthorizeResponse
+    return { data, status }
+  }
+
+  async updateSessionData(
+    paymentSession: {
+      id: string
+      provider_id: string
+      data: Record<string, unknown>
+    },
+    data: Record<string, unknown>
+  ): Promise<Record<string, unknown>> {
+    const provider = this.retrieveProvider(paymentSession.provider_id)
+
+    const res = await provider.updatePaymentData(paymentSession.id, data)
+    if (isPaymentProcessorError(res)) {
+      this.throwFromPaymentProcessorError(res)
+    }
+
+    return (res as PaymentProcessorSessionResponse).session_data
+  }
+
+  async getStatus(payment: {
+    provider_id: string
+    data: Record<string, unknown>
+  }): Promise<PaymentSessionStatus> {
+    const provider = this.retrieveProvider(payment.provider_id)
+    return await provider.getPaymentStatus(payment.data)
+  }
+
+  async capturePayment(paymentObj: {
+    provider_id: string
+    data: Record<string, unknown>
+  }): Promise<Record<string, unknown>> {
+    const provider = this.retrieveProvider(paymentObj.provider_id)
+
+    const res = await provider.capturePayment(paymentObj.data)
+    if (isPaymentProcessorError(res)) {
+      this.throwFromPaymentProcessorError(res)
+    }
+
+    return res as Record<string, unknown>
+  }
+
+  async cancelPayment(payment: {
+    data: Record<string, unknown>
+    provider_id: string
+  }): Promise<void> {
+    const provider = this.retrieveProvider(payment.provider_id)
+
+    const error = await provider.cancelPayment(payment.data)
+    if (isPaymentProcessorError(error)) {
+      this.throwFromPaymentProcessorError(error)
+    }
+  }
+
+  async refundFromPayment(
```

Comment:
question: Do we need the `From`?
```suggestion
  async refundPayment(
```

Reply:
Just kept the old method name for consistency with the current provider service but we can rename